### ISSUE TITLE
Wrong source code encoding of a file in a project causes jedi "get_references" to raise exception

### ISSUE THREAD

**Issue by pterodragon** [davidhalter/jedi#1676]
I'm exploring the cpython source code with jedi
I hit
```
[coc.nvim] Jedi error: Traceback (most recent call last):
  File "completion.py", line 670, in watch
    response = self._process_request(rq)
  File "completion.py", line 643, in _process_request
    script.get_references(line, column), request["id"]
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/jedi/api/helpers.py", line 487, in wrapper
    return func(self, line, column, *args, **kwargs)
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/jedi/api/__init__.py", line 529, in get_references
    return _references(**kwargs)
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/jedi/api/__init__.py", line 523, in _references
    names = find_references(self._get_module_context(), tree_name, scope == 'file')
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/jedi/inference/references.py", line 148, in find_references
    for module_context in potential_modules:
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/jedi/inference/references.py", line 276, in get_module_contexts
_containing_name
    for x in search_in_file_ios(inference_state, file_io_iterator, name,
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/jedi/inference/references.py", line 289, in search_in_file_ios
    m = _check_fs(inference_state, file_io, regex)
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/jedi/inference/references.py", line 176, in _check_fs
    code = python_bytes_to_unicode(code, errors='replace')
  File "/home/???/.pyenv/versions/general38/lib/python3.8/site-packages/parso/utils.py", line 111, in python_bytes_to_unicode
LookupError: unknown encoding: uft-8
```
due to a test file `cpython/Lib/test/bad_coding.py` having a wrong source code encoding

**Comment by pterodragon** [comment]
potential fix is 
```
def _check_fs(inference_state, file_io, regex):
    try:
        code = file_io.read()
    except FileNotFoundError:
        return None
    try: <<<
        code = python_bytes_to_unicode(code, errors='replace') <<<
    except Exception: <<<
        return None <<<
```

**Comment by davidhalter** [comment]
parso version?

**Comment by pterodragon** [comment]
> parso version?

`parso==0.7.0`

**Comment by davidhalter** [comment]
This issue was fixed in parso 0.7.1.
